### PR TITLE
Label sync and Most accessed flows date filters

### PR DESCRIPTION
**Pull request by PeterAlexandre**
## Add Label:
New _Label_ model created, adding an extra _date_ field to store a daily count for each label.
A _periodic task_ was scheduled to obtain the label count once a day.

This change will allow queries filtered for a period of time, which is what we need for the following dashboard:

image

> New params: start_date, end_date, label__uuid.

> Check the postman collection's shared link at the end of this comment, for an example of how to use the Label endpoint.

## Change _MostAccessedFlows_
Add _date_ filters in the most accessed flow listing endpoints. This change is what we need for the following dashboard:

image

> New params: start_date, end_date.

> Check the postman collection's shared link at the end of this comment,, for an example of how to use the new filters.

### Postman Collection
https://www.getpostman.com/collections/a347048bcf9123f6e75a

### CODE REVIEW

File: healthbuddy_backend/rapidpro/views.py [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+from django.utils.dateparse import parse_date
 from rest_framework import viewsets
 from rest_framework.decorators import action
 from rest_framework.generics import ListAPIView
 from rest_framework.permissions import IsAdminUser
 from rest_framework.response import Response
 from rest_framework.views import APIView
 
-from .models import Flow, DailyFlowRuns, DailyGroupCount, DailyChannelCount
+from .models import Flow, DailyFlowRuns, DailyGroupCount, DailyChannelCount, DailyLabelCount
```
Comment (matmsa27) [inline healthbuddy_backend/rapidpro/views.py:12]: organize this imports pls

**Review by matmsa27 — COMMENTED** [review]
Please validate this PR with pep8 or lint.

File: healthbuddy_backend/rapidpro/tasks.py [hunk 2]
```diff
@@ -4,7 +4,24 @@
...
+    Group,
+    DailyGroupCount,
+    DailyChannelCount,
+    Channel,
+    Label,
+    DailyLabelCount,
+)
+
+
+def get_all_results(next_call, headers, *args):
```
Comment (Sandro-Meireles) [inline healthbuddy_backend/rapidpro/tasks.py:19]: The code is very good, congratulations. However i'm in doubt about, why was `*args` declared if it is not being used at any time? Do you think about using it in the future?

Comment (PeterAlexandre) [reply healthbuddy_backend/rapidpro/tasks.py:19]: Thanks!

To be honest, I copied this code snippet from another solution I made. In fact `*args` is not being used in this implementation, so I am going to remove it.

File: healthbuddy_backend/rapidpro/views.py [hunk 3]
```diff
@@ -175,3 +182,20 @@ def filter_queryset(self, queryset):
...
+class DailyLabelCountListView(ListAPIView):
+    queryset = DailyLabelCount.objects.all()
+    model = DailyLabelCount
+    pagination_class = None
+    serializer_class = DailyLabelCountSerializer
+    filterset_fields = ["label__uuid", "label__name", "day"]
+    search_fields = ["label__uuid", "label__name", "day"]
+    ordering_fields = ["label__uuid", "label__name", "day"]
+
+    def filter_queryset(self, queryset):
```
Comment (mldzs) [inline healthbuddy_backend/rapidpro/views.py:209]: In the future we will be able to create a mixin that is responsible for implementing the filter by date range, and in the CBVs we would only need to inherit from that mixin.

File: healthbuddy_backend/rapidpro/views.py [hunk 4]
```diff
@@ -115,7 +116,13 @@ class MostAccessedFlowStatus(APIView):
     authentication_classes = (QueryParamsFixedTokenAuthentication, JWTAuthentication)
 
     def get(self, request, attribute):
-        flows = Flow.objects.all().filter(is_active=True).annotate(
+        query_params = self.request.query_params
+        start_date = parse_date(query_params.get("start_date", "2000-01-01"))
```
Comment (mldzs) [inline healthbuddy_backend/rapidpro/views.py:126]: Is it really necessary to perform this conversion from string to date? I believe that the filter accepts the string as a date

File: healthbuddy_backend/rapidpro/views.py (hunk 3 above)
Comment (PeterAlexandre) [reply healthbuddy_backend/rapidpro/views.py:209]: It's a crucial suggestion. I will probably work on it the next time I work on this project.

File: healthbuddy_backend/rapidpro/views.py (hunk 4 above)
Comment (PeterAlexandre) [reply healthbuddy_backend/rapidpro/views.py]: Solved

**Review by mldzs — APPROVED** [review]

**Review by Sandro-Meireles — APPROVED** [review]

**Review by matmsa27 — APPROVED** [review]

**Pull request merged**